model/src/order.rs

```diff
@@ -56,6 +58,18 @@ impl Order {
             order_creation,
         })
     }
+    pub fn contains_token_from(&self, token_list: &HashSet<H160>) -> bool {
+        token_list
+            .intersection(&hashset!(
+                self.order_creation.buy_token,
+                self.order_creation.sell_token
+            ))
+            .collect::<HashSet<_>>()
+            .is_empty()
+        // The above evaluation is more "symmetric" while the commented one is more precise
+        // token_list.contains(&self.order_creation.buy_token)
+        //     || token_list.contains(&self.order_creation.sell_token)
```

Comment:
While the intersection is slightly nicer, I might actually opt for the commented version of this condition so not to introduce the `maplit` dependency here.

Reply:
I like the commented version because no temporary set is created. Could also write it as
```rust
        [self.order_creation.buy_token, self.order_creation.sell_token]
        .iter()
        .any(|token| token_list.contains(token))
```

Reply:
Cool, I was gonna switch to this anyway! Your suggestion is intriguing, but maybe more cryptic than it needs to be. Lets see what others think and I'll just use the commented code for now.